### PR TITLE
Fix #10

### LINKED ISSUE #10: scroll-target id lookup fails
I have a DOM tree which is, roughly:

```
<x-component>
  <iron-pages>
     <div id="foo">
        <x-scrollable scroll-target="foo">
```

x-scrollable has IronScrollTargetBehavior attached. The lookup of 'foo' fails and I end up with null in the scroll-target property.

Going through the code I see this:

```
      var ownerRoot = Polymer.dom(this).getOwnerRoot();
      this.scrollTarget = (ownerRoot && ownerRoot.$) ?
          ownerRoot.$[scrollTarget] : Polymer.dom(this.ownerDocument).querySelector('#' + scrollTarget);
```

ownerRoot is ending up being a documentFragment which has no $ property. Thus, the conditional in the next line is failing. Looking at the contents of the documentFragment, it seems to represent the x-component node of the DOM tree.

Switching from shady DOM to shadow DOM doesn't have any effect.

Any suggestions?

### PR DESCRIPTION
**Pull request by blasten**
This should fix the issues.

### CODE REVIEW

File: iron-scroll-target-behavior.html [hunk 1]
```diff
@@ -75,8 +75,8 @@
         if (typeof scrollTarget === 'string') {
 
           var ownerRoot = Polymer.dom(this).getOwnerRoot();
-          this.scrollTarget = (ownerRoot && ownerRoot.$) ?
-              ownerRoot.$[scrollTarget] : Polymer.dom(this.ownerDocument).querySelector('#' + scrollTarget);
+          this.scrollTarget = (ownerRoot && ownerRoot.host && ownerRoot.host.$) ?
+              ownerRoot.host.$[scrollTarget] : Polymer.dom(this.ownerDocument).querySelector('#' + scrollTarget);
```
Comment (frankiefu) [inline iron-scroll-target-behavior.html]: Actually we can just use [domHost](https://github.com/Polymer/polymer/blob/master/src/mini/shady.html#L97) here,  e.g.

```
var host = this.domHost;
this.scrollTarget = (host && host.$) ?
    host.$[scrollTarget] : Polymer.dom(this.ownerDocument).querySelector('#' + scrollTarget);
```

**Comment by frankiefu** [comment]
minor nit, otherwise LGTM

File: iron-scroll-target-behavior.html (hunk 1 above)
Comment (blasten) [inline iron-scroll-target-behavior.html]: sweeet

**Comment by blasten** [comment]
fixed @frankiefu

**Comment by frankiefu** [comment]
:+1:

**Pull request merged**